### PR TITLE
Merge Chainer v.1.23.0

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -57,7 +57,10 @@ script:
   - flake8 --config=.flake8.cython
   - autopep8 -r . --global-config .pep8 | tee check_autopep8
   - test ! -s check_autopep8
-  - PYTHONWARNINGS='module::DeprecationWarning' nosetests -a '!gpu' --with-doctest tests/install_tests
-  - READTHEDOCS=True python setup.py develop
+  - PYTHONWARNINGS='ignore::FutureWarning,module::DeprecationWarning' nosetests -a '!gpu,!slow' --with-doctest tests/install_tests
+  - if [[ $TRAVIS_OS_NAME == "linux" ]]; then
+      cd ..;
```
Comment (unnonouno) [inline .travis.yml]: Is this fix required?

**Comment by okuta** [comment]
@unnonouno I fixed.

File: tests/cupy_tests/indexing_tests/test_generate.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import unittest
 
+from chainer import cuda
```
Comment (unnonouno) [inline tests/cupy_tests/indexing_tests/test_generate.py]: Don't use chainer

File: setup.py [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
 
+import os
 from setuptools import setup
+import sys
 
 import cupy_setup_build
 
 
+if sys.version_info[:3] == (3, 5, 0):
+    if not int(os.getenv('CHAINER_PYTHON_350_FORCE', '0')):
```
Comment (unnonouno) [inline setup.py]: CUPY?

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**